static/js/tools/download/page.tsx

```diff
@@ -0,0 +1,370 @@
+                      YYYY or YYYY-MM or YYYY-MM-DD
+                    </div>
+                  </div>
+                </FormGroup>
+              </div>
+            </div>
+            <div className="download-option-section">
+              <div className="download-option-label">Variables</div>
+              {_.isEmpty(selectedOptions.selectedStatVars) ? (
+                "None selected"
+              ) : (
+                <div className="download-sv-chips">
+                  {Object.keys(selectedOptions.selectedStatVars).map((sv) => {
+                    return (
+                      <Chip
+                        key={sv}
+                        id={sv}
+                        title={selectedOptions.selectedStatVars[sv].title || sv}
+                        removeChip={removeSV}
+                      />
+                    );
+                  })}
+                </div>
+              )}
+            </div>
+            <Row className="d-lg-none">
+              <Col>
+                <Button color="primary" onClick={toggleSvModalCallback}>
+                  Select variable
+                </Button>
+              </Col>
+            </Row>
+            <Button onClick={onGetDataButtonClicked} color="primary">
+              {getDataButtonText}
+            </Button>
+          </PlaceSelector>
+        </div>
+        {!_.isEmpty(validationErrors.incompleteSelectionMessage) && (
+          <div>{validationErrors.incompleteSelectionMessage}</div>
+        )}
+        {showPreview && (
+          <div>{previewDisabled ? "disabled preview" : "preview"}</div>
+        )}
+        {showInfo && <div>info text</div>}
+      </div>
+    </>
+  );
+
+  function selectSV(sv: string, svInfo: StatVarInfo): void {
+    setSelectedOptions((prev) => {
+      const updatedSV = _.cloneDeep(prev.selectedStatVars);
+      updatedSV[sv] = svInfo;
+      return { ...prev, selectedStatVars: updatedSV };
+    });
+  }
+
+  function removeSV(sv: string): void {
+    setSelectedOptions((prev) => {
+      const updatedSV = _.cloneDeep(prev.selectedStatVars);
+      if (sv in updatedSV) {
+        delete updatedSV[sv];
+      }
+      return { ...prev, selectedStatVars: updatedSV };
+    });
+  }
+
+  function getOptionsFromURL(): void {
+    const urlParams = new URLSearchParams(window.location.hash.split("#")[1]);
+    const place = urlParams.get(URL_PARAM_KEYS.PLACE);
+    const placePromise = place
+      ? getNamedTypedPlace(place)
+      : Promise.resolve({ dcid: "", name: "", types: null });
+    const statVarsParam = urlParams.get(URL_PARAM_KEYS.STAT_VARS);
+    const statVarsList = statVarsParam ? statVarsParam.split(SEPARATOR) : [];
+    const svInfoPromise = !_.isEmpty(statVarsList)
+      ? getStatVarInfo(statVarsList)
+      : Promise.resolve({});
+    const enclosedPlaceType = urlParams.get(URL_PARAM_KEYS.PLACE_TYPE) || "";
+    const dateRange =
+      urlParams.get(URL_PARAM_KEYS.DATE_RANGE) === "1" ? true : false;
```

Comment:
constant: done

code simplification: not sure how this can be line 264-278 and line 304-317 would be able to use a common url parse function because line 304-317 is not parsing the url?

Reply:
ah, you're right. i do like the simplification with the options object on line 265 - it makes it clear what the defaults are too.